web/controllers/password_reset_controller.ex

```diff
@@ -11,14 +11,14 @@ defmodule CodeCorps.PasswordResetController do
   and return email. 422 if pwd do not match or auth token does not exist
   """
   def reset_password(conn, %{"token" => token, "password" => password, "password_confirmation" => password_confirmation}) do
-    user = conn.assigns.current_user
-    with %AuthToken{value: auth_token} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token, user_id: user.id }),
+    with %AuthToken{value: auth_token, user_id: user_id} <- Repo.get_by(CodeCorps.AuthToken, %{ value: token }),
       {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", auth_token, max_age: 1209600) do
-        with %Changeset{valid?: true} <- User.reset_password_changeset(user, 
-                                                                       %{password: password, password_confirmation: password_confirmation}) do
+        user = Repo.get(User, user_id)
+        with %Changeset{valid?: true} = changeset <- User.reset_password_changeset(user, 
+                                                                       %{password: password, password_confirmation: password_confirmation}),
+             {:ok, user} <- Repo.update(changeset) do
           conn
-          |> put_status(:created)
-          |> render("show.json", email: user.email)
+          |> TokenController.create(%{"username" => user.email, "password" => password})
```

Comment:
@snewcomer The two with blocks are really hard to follow here.. The code could be rewritten, and IMO would be easier to follow as: 

```Elixir
def reset_password(conn, %{"token" => reset_token, "password" => password, "password_confirmation" => password_confirmation} = params) do
  with %AuthToken{user: user} <- AuthToken |> Repo.get_by(%{ value: reset_token }) |> Repo.preload(:user),
       {:ok, _} <- Phoenix.Token.verify(CodeCorps.Endpoint, "user", reset_token, max_age: 1209600),
       {:ok, updated_user} <- user |> User.reset_password_changeset(params) |> Repo.update,
       {:ok, auth_token, _claims} = updated_user |> Guardian.encode_and_sign(:token)
  do
    conn
    |> Plug.Conn.assign(:current_user, updated_user)
    |> put_status(:created)
    |> render("show.json", token: auth_token, user_id: updated_user.id, email: updated_user.email)
  else
    {:error, %Changeset{} = changeset} -> conn |> put_status(422) |> render(:errors, data: changeset)
    nil -> conn |> put_status(:not_found) |> render(CodeCorps.ErrorView, "404.json")
  end
end
```